cmd/promote/git/app_interface.go

```diff
@@ -46,24 +46,29 @@ func replaceTargetSha(fileContent string, targetSuffix string, promotionGitHash
 		return "", fmt.Errorf("error parsing saas YAML: %v", err), false
 	}
 	targetFound := false
-	targets, err := kyaml.Lookup("resourceTemplates", "[path="+resourcePath+"]", "targets").Filter(node)
+	rts, err := kyaml.Lookup("resourceTemplates").Filter(node)
 	if err != nil {
-		return "", fmt.Errorf("error querying saas YAML: %v", err), false
-	}
-	err = targets.VisitElements(func(element *kyaml.RNode) error {
-		name, _ := element.GetString("name")
-		match, _ := regexp.MatchString("(.*)"+targetSuffix, name)
-		if match {
-			targetFound = true
-			fmt.Println("updating target: ", name)
-			_, err = element.Pipe(kyaml.SetField("ref", kyaml.NewStringRNode(promotionGitHash)))
-			if err != nil {
-				return fmt.Errorf("error setting ref: %v", err)
-			}
+		return "", fmt.Errorf("error querying resource templates: %v", err), false
+	}
+	for i := range len(rts.Content()) {
+		targets, err := kyaml.Lookup("resourceTemplates", strconv.Itoa(i), "targets").Filter(node)
+		if err != nil {
+			return "", fmt.Errorf("error querying saas YAML: %v", err), false
 		}
-		return nil
-	})
-
+		err = targets.VisitElements(func(element *kyaml.RNode) error {
+			name, _ := element.GetString("name")
```

Comment:
Naive question (and I realize this isn't your code and existed in the previous version) - is there a good reason to ignore the error here? 

It seems like it was deliberate because the regexp.MatchString also ignores the error, which would report of the pattern was invalid.  So, maybe there's not _always_ a name?  :thinking: